WcaOnRails/spec/factories/persons.rb

```diff
@@ -3,9 +3,9 @@
   factory :person do
     wca_id do
       mid = ('A'..'Z').to_a.sample(4).join
-      "2016#{mid}01".tap do |id|
-        id.next! while Person.exists?(wca_id: id)
-      end
+      id = "2016#{mid}01"
+      id = id.next while Person.exists?(wca_id: id)
+      id
```

Comment:
Not sure it's an improvement, but this is as far as I can squeeze this one:

    wca_id do
      mid = ('A'..'Z').to_a.sample(4).join
      num = 0
      begin id = "2016#{mid}0#{num += 1}" end while Person.exists?(wca_id: id)
      id
    end